core/ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/app/AppContext.java

```diff
@@ -84,22 +175,21 @@
     void setStartUpActions(List<StartUpAction> startUpActions);
 
     /**
-     * Returns {@link Factory} instance which id was set on startup,
-     * or {@code null} if no factory was specified.
+     * Returns {@link Factory} instance which id was set on startup, or {@code null} if no factory was specified.
      *
      * @return loaded factory or {@code null}
      */
     Factory getFactory();
 
-    void setFactory(Factory factory);
-
-    /** Returns instance  of the developer machine (where workspace is bound). */
-    DevMachine getDevMachine();
-
-    void setDevMachine(DevMachine devMachine);
+    String getWorkspaceId();
 
-    String getProjectsRoot();
+    /* Deprecated methods */
 
-    void setProjectsRoot(String projectsRoot);
+    /* No more supported, will be removed soon. */
```

Comment:
either it is supported but deprecated or it has to be removed
also it's great when describing/commenting deprecated methods, to tell/link to what method should be used instead